### PR TITLE
feat: `sway-fmt-v2` preserves newline with given threshold

### LINKED ISSUE #2379: `sway-fmt-v2` should be able to preserve whitespace
Currently all newlines aren't handled, so something like:
```rust
let foo = "foo";

let bar = "bar";

let baz = "baz";
```

will format to: 
```rust
let foo = "foo";
let bar = "bar";
let baz = "baz";
```

I think we could add some kind of threshold with the default being 2, that keeps track of newlines and inserts them similar to how we handle inserting comments.

### PR DESCRIPTION
**Pull request by kayagokalp**
~Opening up for visibility.~
closes #2379.

## TODO:

- [x] Move threshold to config
- [x] Add comma

### CODE REVIEW

**Comment by eureka-cpu** [comment]
Huge 🙌🏻 Very excited to get this in

File: sway-fmt-v2/src/fmt.rs [hunk 1]
```diff
@@ -855,4 +863,37 @@ impl Qux for Foo {
...
+    let number3: u64 = 30;
+
+
+
+}"#;
+
+        let correct_sway_code = r#"script;
+
+fn main() {
+    let number: u64 = 10;
```
Comment (eureka-cpu) [inline sway-fmt-v2/src/fmt.rs:887]: Wouldn't we want to retain the extra space after this line? and if there are more lines than the threshold those extra lines would be removed?

**Review by eureka-cpu — APPROVED** [review]
Fantastic work, thanks so much @kayagokalp !

**Review by JoshuaBatty — APPROVED** [review]
:shipit:

**Pull request merged**